packages/flutter_driver/lib/src/android/common.dart

```diff
@@ -0,0 +1,187 @@
+
+  /// Whether the node is editable.
+  ///
+  /// This is usually only applied to text fields, which map
+  /// to "android.widget.EditText".
+  bool get isEditable => _flags['isEditable'];
+
+  /// Whether the node is enabled.
+  bool get isEnabled => _flags['isEnabled'];
+
+  /// Whether the node is focusable.
+  bool get isFocusable => _flags['isFocusable'];
+
+  /// Whether the node is focused.
+  bool get isFocused => _flags['isFocused'];
+
+  /// Whether the node represents a password field.
+  ///
+  /// Equivalent to [SemanticsFlag.isObscured].
+  bool get isPassword => _flags['isPassword'];
+
+  /// Whether the node is long clickable.
+  ///
+  /// Equivalent to having [SemanticsAction.longPress]/
+  bool get isLongClickable => _flags['isLongClickable'];
+
+  /// Gets a [Rect] which defines the position and size of the semantics node.
+  Rect getRect() {
+    final Map<String, Object> rawRect = _values['rect'];
+    final Map<String, int> rect = rawRect.cast<String, int>();
+    return new Rect.fromLTRB(
+      rect['left'].toDouble(),
+      rect['top'].toDouble(),
+      rect['right'].toDouble(),
+      rect['bottom'].toDouble(),
+    );
+  }
+
+  /// Gets a [Size] which defines the size of the semantics node.
+  Size getSize() {
+    final Rect rect = getRect();
+    return new Size(rect.bottom - rect.top, rect.right - rect.left);
+  }
+
+  /// Gets a list of [AndroidSemanticsActions] which are defined for the node.
+  List<AndroidSemanticsAction> getActions() {
+    final List<AndroidSemanticsAction> result = <AndroidSemanticsAction>[];
+    for (int id in _values['actions']) {
+      result.add(AndroidSemanticsAction.deserialize(id));
+    }
+    return result;
+  }
+
+  @override
+  String toString() {
+    return _values.toString();
+  }
+}
+
+
+/// A Dart VM implementation of a rectangle.
+///
+/// Created to mirror the implementation of `Rect` from 'dart:ui';
+@immutable
+class Rect {
+  /// Creates a new rectangle.
+  ///
+  /// All values are required.
+  const Rect.fromLTRB(this.left, this.top, this.right, this.bottom);
+
+  /// The top side of the rectangle.
+  final double top;
+
+  /// The left side of the rectangle.
+  final double left;
+
+  /// The right side of the rectangle.
+  final double right;
+
+  /// The bottom side of the rectangle.
+  final double bottom;
+
+  @override
+  int get hashCode =>
+      top.hashCode ^ left.hashCode ^ right.hashCode ^ bottom.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    final Rect typedOther = other;
+    return typedOther.top == top &&
+        typedOther.left == left &&
+        typedOther.right == right &&
+        typedOther.bottom == bottom;
+  }
+
+  @override
+  String toString() => 'Rect.fromLTRB($left, $top, $right, $bottom)';
+}
+
+/// A Dart VM implementation of a Size.
+///
+/// Created to mirror the implementation of `Size` from 'dart:ui'
```

Comment:
here and elsewhere: you can use `[ui.Size]` to get this to cross-reference correctly, I believe.

Reply:
(even though it's not in scope here - dartdocs doesn't care, it puts everything in scope)